src/fastcs/backends/rest/rest.py

```diff
@@ -0,0 +1,147 @@
+from collections.abc import Awaitable, Callable, Coroutine
+from dataclasses import dataclass
+from types import MethodType
+from typing import Any
+
+import uvicorn
+from fastapi import FastAPI
+from pydantic import create_model
+
+from fastcs.attributes import AttrR, AttrRW, AttrW, T
+from fastcs.controller import BaseController
+from fastcs.mapping import Mapping
+
+
+@dataclass
+class RestServerOptions:
+    host: str = "localhost"
+    port: int = 8080
+    log_level: str = "info"
+
+
+class RestServer:
+    def __init__(self, mapping: Mapping):
+        self._mapping = mapping
+        self._app = self._create_app()
+
+    def _create_app(self):
+        app = FastAPI()
+        _add_dev_attributes(app, self._mapping)
+        _add_dev_commands(app, self._mapping)
+
+        return app
+
+    def run(self, options: RestServerOptions | None = None) -> None:
+        if options is None:
+            options = RestServerOptions()
+
+        uvicorn.run(
+            self._app,
+            host=options.host,
+            port=options.port,
+            log_level=options.log_level,
+        )
+
+
+def _put_request_body(attribute: AttrW[T]):
+    return create_model(
+        f"Put{str(attribute.datatype.dtype)}Value",
+        **{"value": (attribute.datatype.dtype, ...)},  # type: ignore
```

Comment:
This could do with some explanation. And the two similar functions below.

Why is the key `value`? What does `(attribute.datatype.dtype, ...)` do? Maybe just a case of referencing FastAPI docs?

Reply:
> Why is the key `value`? 

My intention was a schema defined for the request body for each datatype to have the least possible schemas. An alternative could be a schema with the attribute as a key, but thats potentially a lot of schemas that are the same and less navigable in `/docs`.

I choose "value" because im not aware of standards for this but it seems fairly common (as per https://learn.microsoft.com/en-us/azure/azure-app-configuration/rest-api-key-value?pivots=v1). I dont want to use "data" because that seems to be often used as an object ( as per https://developer.hashicorp.com/vault/api-docs).

> What does `(attribute.datatype.dtype, ...)` do?

This creates the schema without defaults. I will add a comment

Reply:
Also, I think you can remove the `# type: ignore` if you do 

```
return create_model(
    f"Put{str(attribute.datatype.dtype)}Value",
    value=(attribute.datatype.dtype, ...),
)
```

?

Reply:
> Also, I think you can remove the `# type: ignore` if you do
> 
> ```
> return create_model(
>     f"Put{str(attribute.datatype.dtype)}Value",
>     value=(attribute.datatype.dtype, ...),
> )
> ```
> 
> ?

This is true - Thank you

Reply:
I have noticed that `f"Put{str(attribute.datatype.dtype)}Value"` renders out as `Put<class 'bool'>Value`. I think if I were to be consistent with my intention of a schema per datatype I should aim to create `PutBoolValue`